xc7/utils/prjxray_edge_library.py

```diff
@@ -688,6 +692,28 @@ def connect_at(
                     yield (src_node, switch_pkey, dest_node, pip.pip_pkey)
                     return
 
+        elif self.pins and other_connector.pins and pip.is_pseudo:
+            switch_pkey = pip.get_pip_switch(src_wire_pkey, dest_wire_pkey)
+
+            for pin_dir in self.pins.edge_map:
+                if pin_dir in other_connector.pins.edge_map:
+                    src_node = self.pins.edge_map[pin_dir]
+                    dest_node = other_connector.pins.edge_map[pin_dir]
+
+                    src_wire_switch_pkey, src_wire_node = self.find_wire_node(
```

Comment:
In the same way the edge to channel or channel to edge should be CHAN to IPIN or OPIN to CHAN, let's assert that this src_node is an IPIN and dest_node is an OPIN.

Reply:
Done